### PR TITLE
Modern eslint ES5 compatibility

### LINKED ISSUE #50: Tiny issue -- readme sample eslint config uses "config" option key instead of correct "configFile" option key.
Reference:
http://eslint.org/docs/developer-guide/nodejs-api#cliengine

That's it :)

### LINKED ISSUE #106: .eslintrc config out of date for `space-after-keyword` ?
Using npm package  -g eslint v2.8.0

Project with "mobify-code-style": "^2.4.3"

untitled

Replacing line 72 for "keyword-spacing" seems to work well. Anyone else experiencing this issue?

### LINKED ISSUE #115: Missing eslint v2 and v3 compatible config files
Directly related to #106 , and with v3 of eslint being using by the Progressive Web SDK, we are currently lacking a v2 and v3 compatible eslint config file in the repo.

### PR DESCRIPTION
**Pull request by DanielBrookRoberge**
## Changes
- Add an ESLint 2+ compatible version of the ES5 linting standard in the `es5` directory
- Modify the instructions to use it.
- Fixes #115 
- Fixes #106 
- Fixes #50 

## How To Test
- Update `grunt-eslint` in a repository to the latest version
- Follow the migration instructions in the new README.md
- Run the linter and see that it works.

## Applicable Research Resources
- (links, optional)

## TODOs:
- [x] +1
- [x] Updated README
- [x] Updated CHANGELOG

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,20 +36,36 @@ eslint:{
             // When true, eslint will test _only_ the rules set in the provided
             // configuration file
             reset: false,
-            config: 'node_modules/mobify-code-style/javascript/.eslintrc'
+            config: './.eslintrc.yml'
```
Comment (marlowpayne) [inline README.md]: Let's fix #50 while we're here and update the key name to `configFile`

File: es5/mobify-es5.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  max-depth:
+    - error
+    - max: 5
+  max-len:
+    - error
+    - code: 300
+      tabWidth: 4
+  semi: error
+  func-style: error
+  comma: error
```
Comment (marlowpayne) [inline es5/mobify-es5.yml]: While testing this new config, I'm getting a `Definition for rule 'comma' was not found` error. It also doesn't appear to exist in the [rules doc](http://eslint.org/docs/rules/).

Comment (DanielBrookRoberge) [inline es5/mobify-es5.yml]: oops, that's supposed to be `comma-style`

**Comment by marlowpayne** [comment]
Just want to point out that extending from `eslint:recommended` will enable the [no-unused-vars](http://eslint.org/docs/rules/no-unused-vars) rule. This is a bit of an issue with previous Adaptive builds, as discussed in #78

**Comment by marlowpayne** [comment]
Also pointing out that `eslint:recommended` turns on the [no-console](http://eslint.org/docs/rules/no-console) rule.

**Comment by DanielBrookRoberge** [comment]
I'll just remove `eslint:recommended` since for any new project we probably want to use ES6 anyways.

**Comment by marlowpayne** [comment]
Thanks for these changes @DanielBrookRoberge ! Tested on a couple Adaptive projects. 👍

**Pull request merged**